libpod/runtime_ctr.go

```diff
@@ -246,6 +249,40 @@ func (r *Runtime) newContainer(ctx context.Context, rSpec *spec.Spec, options ..
 }
 
 func (r *Runtime) setupContainer(ctx context.Context, ctr *Container) (_ *Container, retErr error) {
+	if ctr.IsDefaultInfra() || ctr.IsService() {
+		tmpDir, err := r.TmpDir()
+		if err != nil {
+			return nil, fmt.Errorf("getting runtime temporary directory: %w", err)
+		}
+		tmpDir = filepath.Join(tmpDir, "infra-containers", ctr.ID())
```

Comment:
Do we really care about creating a unique tmp dire per container? Seems like we just create inodes for no benifit?
I think one empty dir should be enough as the overlayfs code should take care of the unique dirs.

Reply:
@Luap99 , do you mean  just sharing single rootfs directory across multiple containers? I was thinking there might be an issue with binary changing while the container is running.

The first container can create this rootfs with catatonit-1.0, then catatonit is upgraded to 1.1 and next container will also update the rootfs (it would probably also have to umount the catatonit-1.0 and mount catatonit-1.1 there) which would change things on-the-fly for the first container.

it's probably not an issue since the catatonit is already running in the first container, so maybe it is fine.

Reply:
Yes a single shared directory is fine. Mounting wise it doesn't matter, the container has a) private overlayfs so they do not actually end up mounted in the same place and b) the container has a private mount propagation so the catatonit mount private in each container. You don't see it on the host at all as such there cannot be conflicts with that either.